### ISSUE TITLE
Is CompileTimeValidate() supported?

### ISSUE THREAD

**Issue by GeirLudvigsen** [vescon/MethodBoundaryAspect.Fody#114]
Hi. 

I have written a multicasting logging aspect inheriting OnMethodBoundaryAspect. 
I have declared this aspect on class level where I want it weaved.
`    [TracerAspect(AttributeTargetMemberAttributes = MulticastAttributes.Public | MulticastAttributes.Internal | MulticastAttributes.Protected)]
public class MyClass
    {}`

I was hoping I could override the `CompileTimeValidate(MethodBase method)` to write my own rules for deciding which methods to weave or not. But even if I always return false from CompileTimeValidate() the aspect is still weaved into all methods. 

`public override bool CompileTimeValidate(MethodBase method)
{
      return false;
 }
`

**Comment by Ralf1108** [comment]
Hi GeirLudvigsen,

unfortunately this feature is not implemented yet. I exists in PostSharp so the method was added.
As it has no functionality now it could also be removed

Sorry for that but if you have the knowledge how to implement it we accept PRs ;-)

**Comment by GeirLudvigsen** [comment]
Thanks for your answer!

I found out that implementing CompileTimeValidate() would require System.Reflection and during compile time creating an instance of both the Attribute and the weaving method's class. I guess that could be a real performance hog, so I didn't explore it more.

In stead I added the possibility to filter which methods to weave by assigning Regex patterns for namespace, typename and/or methodname. This will cover my needs. I added a pull request for it now.
